# Bonus: Temperature Analysis I

In [15]:
import pandas as pd
from datetime import datetime as dt

In [21]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [22]:
df.tail()

station        date  prcp  tobs
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

In [23]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [24]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])

In [25]:
#Drop date column
df = df.drop(columns = 'date')

In [27]:
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [29]:
from scipy import stats


In [28]:
# Filter data for desired months
june_data=df[df.index.month==6]
december_data=df[df.index.month==12]

In [32]:
# Identify the average temperature for June
june_data.mean()


prcp     0.136360
tobs    74.944118
dtype: float64

In [33]:
june_data.describe()

prcp         tobs
count  1574.000000  1700.000000
mean      0.136360    74.944118
std       0.335731     3.257417
min       0.000000    64.000000
25%       0.000000    73.000000
50%       0.020000    75.000000
75%       0.120000    77.000000
max       4.430000    85.000000

In [34]:
# Identify the average temperature for December
december_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [35]:
december_data.describe()

prcp         tobs
count  1405.000000  1517.000000
mean      0.216819    71.041529
std       0.541399     3.745920
min       0.000000    56.000000
25%       0.000000    69.000000
50%       0.030000    71.000000
75%       0.150000    74.000000
max       6.420000    83.000000

In [39]:
# Create collections of temperature data
june_temp = june_data.tobs
december_temp = december_data.tobs

In [38]:
june_temp.head()

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
Name: tobs, dtype: int64

In [40]:
december_temp.head()

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
Name: tobs, dtype: int64

In [43]:
# Run paired t-test
stats.ttest_ind(june_temp,december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The mean temperatures found across all stations in June and December for the years 2010-2017 change by around 3.9 degrees F. When a paired t test was run, it had a high p-value of 3.90521, which deems is not statistically significant, therefore indicates strong evidence of a null hypothesis (meaning there exists no relationship in our data).